File: 19th/코딩테스트 기초반/week2/4673번_셀프 넘버.cpp
#include <iostream>

using namespace std;

// 전역 변수에 선언하면 모든 값이 false로 초기화 됨
bool no_self_num[10001]; 

int main() {
    for (int i = 1; i <= 10000; i++) {
        int num = i;
        int sum = num;
        
        // num이 0이 아니라면 반복문 실행
        while (num) { 
            sum += num % 10; // num을 10으로 나눈 나머지, 즉 현재 num의 일의 자리 수
            num /= 10; // num을 10으로 나눈 몫
        }
        
        // sum이 10000 이하일 경우
        if (sum <= 10000) {
           no_self_num[sum] = true; 
        }
    }
    
    for (int i = 1; i <= 10000; i++) {
        // self_num인 값들만 출력
        if (no_self_num[i] == false) cout << i << "\n";
    }
}

File: 20th/코딩테스트 기초반/week2/4673번_셀프 넘버.py
n = 1
arr = []

for i in range(1, 10001):
    num = i + sum(int(j) for j in str(i))  # 숫자 i를 문자열로 만들어 각 자릿수를 sum하고 i를 더함
    arr.append(num)  # 그 수를 배열에 저장함

    if i in arr:  # i를 1부터 10000까지 순회하면서 i가 배열 안에 없을 때만 i를 출력함
        continue
    else:
        print(i)